packages/session-replay-browser/src/session-replay.ts

```diff
@@ -281,14 +274,37 @@ export class SessionReplay implements AmplitudeSessionReplay {
     }
 
     if (this.shouldOptOut()) {
-      this.loggerProvider.log(`Opting session ${this.identifiers.sessionId} out of recording due to optOut config.`);
+      this.loggerProvider.log(
+        `Opting session ${this.identifiers.sessionId} out of replay capture due to optOut config.`,
+      );
       return false;
     }
 
-    const isInSample = isSessionInSample(this.identifiers.sessionId, this.config.sampleRate);
-    if (!isInSample) {
-      this.loggerProvider.log(`Opting session ${this.identifiers.sessionId} out of recording due to sample rate.`);
-      return false;
+    // If targetingConfig exists, we'll use the sessionTargetingMatch to determine whether to record
+    // Otherwise, we'll evaluate the session against the overall sample rate
+    if (this.config.targetingConfig) {
+      if (!this.sessionTargetingMatch) {
+        this.loggerProvider.log(
+          `Not capturing replays for session ${this.identifiers.sessionId} due to not matching targeting conditions.`,
+        );
+        return false;
+      } else {
+        this.loggerProvider.log(
+          `Capturing replays for session ${this.identifiers.sessionId} due to matching targeting conditions.`,
+        );
+        return true;
+      }
```

Comment:
WDYT about this?

```suggestion
      }
      this.loggerProvider.log(
        `Capturing replays for session ${this.identifiers.sessionId} due to matching targeting conditions.`,
      );
```

It seems more inline with the else which only returns early if it needs to.